app/src/main/kotlin/compiler/ast/AstNode.kt

```diff
@@ -28,6 +28,8 @@ sealed interface AstNode {
                 is Function -> "czynność ${this.name}(${this.parameters.joinToString { it.toSimpleString() }}) -> ${this.returnType}"
                 is Function.Parameter -> "${this.name}: ${this.type}${if (this.defaultValue != null) " = ${this.defaultValue.toSimpleString()}" else ""}"
                 is Expression.FunctionCall.Argument -> "${if (this.name != null) "${this.name} = " else ""}${this.value.toSimpleString()}"
+
+                else -> {}
```

Comment:
Are we sure this is desired? Because of this, adding a new type without addressing it in this `when` will no longer fail to compile.

Reply:
So, I did two things: I made `Program` an `AstNode` and replaced `Any`s with `AstNode`s in some places. Since there was case checking on `Any` (!) and it became case checking on `AstNode`, Kotlin *then* started complaining that the checks are not exhaustive, so I added these `else -> {}`, but I got carried away and also added them in `AstNode`.

Reply:
Fixed.